File: codeForces/Problem165b.py
import sys

def check(n, k, v):
	i = 0
	lines = 1
	while(lines!=0):
		lines = v // (k**i)
		n -= lines
		#print(n)
		if (n <= 0):
			return True
		i += 1

	return False

def times(n, k):
	low = 1
	hi = n

	while hi >= low:
		mid = (hi + low) // 2
		
		#checks if mid works
		if check(n, k, mid):
			hi = mid
		#checks if low works
		elif check(n, k, low):
			print(low)
			return
		#if mid and low dont work, checks if hi is the answer 
		elif check(n, k, hi - 1) == False:
			print(str(hi))
			return
		#if hi - 1 works then low becomes mid 
		else:	
			low = mid

def main():
	for line in sys.stdin:
		x = [int(a) for a in line.split(" ")]
		times(x[0], x[1])

main()

File: codeForces/Problem165b.java
import java.util.*;

public class Problem165b{

	public static boolean check(int n, int k, int v){
		int i = 0;
		int lines = 1;
		while(lines != 0){
			lines = v / (int)(Math.pow(k, i));
			n -= lines;
			if(n<=0)
				return true;
			i++;
			if(i > 9999999)
				return false;
		}
		return false;
	}

	public static void times(int n, int k){
		int lo = 1;
		int hi = n;
		if(hi == lo){
			System.out.println(hi);
			return;
		}
		
		while(hi >= lo){
			int mid = lo + (hi - lo) / 2;
			if(check(n, k, mid)){
				hi = mid;
			}
			else if(check(n, k, lo)){
				System.out.println(lo);
					return;
			}
			else if(!check(n, k, hi-1)){
				System.out.println(hi);
				return;
			}
			else
				lo = mid;
		}
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int k = sc.nextInt();
		times(n, k);


	}
}